### ISSUE TITLE
Remove specific dependency in field_divB0

### ISSUE THREAD

**Issue by krystophny** [itpplasma/GORILLA#21]
https://github.com/itpplasma/GORILLA/blob/4cfe4071893e84672460b9d2695458d5ef777a74/SRC/field_divB0.f90#L67

This makes it hard to decouple code and leads to "yet another variant of field_divB0.f90". Check also similar things elsewhere.

**Comment by krystophny** [comment]
possible solution: add namelist section specifically for `field_divB0` members and move the variables into a module inside `field_divB0.f90`.

**Comment by krystophny** [comment]
Oh no it gets worse. Any workaround I tried to at least set the variables is prohibited by the well-meant `protected` flag.

https://github.com/itpplasma/GORILLA/blob/2d54b351d42ba8c146f4f324861dc977f3f8dcdc/SRC/tetra_grid_settings_mod.f90#L27

**Comment by krystophny** [comment]
*Automated triage pass (Claude), grounded in the repository at `210dffd`. Verify before acting.*

Still valid. The dependency is unchanged on `main`, the line numbers have only shifted (the Feb-2025 modularization, commits `db53f83`/`069518d`, wrapped the file in `field_divB0_mod`), and libneo has meanwhile grown the exact factored-out form that GORILLA should converge to. Spec below.

## Purpose

`SRC/field_divB0.f90` is a copy of libneo's `src/magfie/field_divB0.f90` — the shared divergence-free field evaluator used across the NEO-2 lineage. GORILLA's copy has been patched to reach *upward* into `tetra_grid_settings_mod`, a GORILLA-specific namelist reader. That inverts the dependency direction: a shared low-level field module now cannot compile or run without GORILLA's grid-settings namelist, which is precisely why the file has to be forked rather than shared, and why every consumer accumulates its own variant. The fix is to invert the flow — GORILLA *pushes* its filenames into the field layer, the field layer knows nothing about GORILLA — which also removes the `protected` blocker raised in the second comment, because nothing needs to write into `tetra_grid_settings_mod` any more.

## Current state

The offending import, `SRC/field_divB0.f90:28` (the permalink in the issue body, post-modularization):

```fortran
  use tetra_grid_settings_mod, only: g_file_filename, convex_wall_filename, iaxieq_in !=> Michael Eder, 05 May 2022
```

and the read/assign block it feeds, `SRC/field_divB0.f90:42-67`. Line 44 opens the file, lines 45-58 read the 13 positional entries (three of them into a throwaway `dummy`), and lines 63-66 overwrite the discarded ones:

```fortran
  if(icall .eq. 0) then
     icall = 1
     open(newunit=iunit, file='field_divB0.inp')
     ...
     read(iunit,*) dummy !gfile        ! equilibrium file => Michael Eder, 08 March 2021
     read(iunit,*) pfile        ! coil        file
     read(iunit,*) dummy !convexfile   ! convex file for stretchcoords => Michael Eder, 08 March 2021
     ...
     read(iunit,*,err=7) dummy !iaxieq
7    close(iunit)
     ...
     !Use input data from tetra_grid_settings_mod => Michael Eder, 05 May 2022
     gfile = g_file_filename
     convexfile = convex_wall_filename
     iaxieq = iaxieq_in
  endif
```

Targets of the assignments: `gfile`, `convexfile` live in `module input_files` (`SRC/utils_bdivfree.f90:1-9`), `iaxieq` in `module field_mod` (`SRC/bdivfree.f90:55-60`). Consumers:

* `gfile` → `read_dimeq1` (`SRC/utils_bdivfree.f90:842-857`) and `read_eqfile1` (`SRC/utils_bdivfree.f90:874-933`), plus the WEST pair `read_dimeq_west` (`SRC/utils_bdivfree.f90:962-974`) / `read_eqfile_west`.
* `convexfile` → `stretch_coords` (`SRC/bdivfree.f90:974-1054`; the `open(newunit=iunit,file=trim(convexfile))` is at line 993).
* `iaxieq` → the `select case(iaxieq)` in `field_eq` at `SRC/bdivfree.f90:1107-1140` (`use field_mod, only : iaxieq` at line 1091).

The blocker from the second comment is real and unchanged: every module variable in `SRC/tetra_grid_settings_mod.f90:9-67` is declared `public, protected`, e.g. `SRC/tetra_grid_settings_mod.f90:32-33`:

```fortran
    character*64,public,protected :: g_file_filename
    character*64,public,protected :: convex_wall_filename
```

Reference implementation to converge on — libneo (`itpplasma/libneo`, `main`) has already extracted the reader with an optional path and **no** external settings dependency, in `src/magfie/field_divB0.f90` (module `field_sub`):

```fortran
subroutine read_field_input(input_file)
  use input_files, only : iunit, gfile, pfile, convexfile, fluxdatapath, ieqfile
  use field_c_mod, only : ntor, icftype
  use field_eq_mod, only : nwindow_r, nwindow_z
  use field_mod, only : ipert, iequil, ampl
  use inthecore_mod, only : cutoff

  character(*), intent(in), optional :: input_file

  if (present(input_file)) then
    open(iunit, file=input_file, status='old', action='read')
  else
    open(iunit, file='field_divB0.inp', status='old', action='read')
  end if
  ...
```

Note the format has already drifted: libneo carries `ieqfile` in `input_files` (`libneo/src/magfie/input_files.f90:4`, `integer :: ieqfile=1 ! equilibrium file type (0 - old, 1 - GEQDSK, 2 - WEST GEQDSK)`), while GORILLA carries `iaxieq` in `field_mod` (`0` EFIT / `1` WEST). Same slot in the file, different name, different meaning of the values. That divergence is the concrete cost the issue predicted.

The pattern the fix should follow **already exists in this repo** for the analytic circular tokamak field added in May 2026 (commit `561844c`) — `SRC/tetra_grid_mod.f90:138-141` pushes state into the field layer instead of the field layer pulling it:

```fortran
          case(5) !analytic circular tokamak
            ianalytic_circ = 1
            call set_field_analytic_circ(R0_analytic_circ, a_analytic_circ, &
                                         B0_analytic_circ, q0_analytic_circ, q1_analytic_circ)
```

(`ianalytic_circ` here is `field_mod`'s, imported at `SRC/tetra_grid_mod.f90:37`; the setter is `SRC/field_analytic_circ_mod.f90:27-38`.) Generalize that to `gfile`/`convexfile`/`iaxieq`.

"Similar things elsewhere" — two more instances of the same inversion:

* `SRC/canonical_coordinates_mod.f90:11`: `module new_vmec_stuff_mod` opens with `use tetra_grid_settings_mod, only: netcdffile => netcdf_filename`. libneo's counterpart declares it as an ordinary settable variable, `libneo/src/vmec_support/canonical_coordinates_mod.f90:12`: `character*1000   :: netcdffile = ''  ! Empty = unset, apply 'wout.nc' default at usage`.
* `SRC/field_line_integration_for_SYNCH.f90:30`: `use tetra_grid_settings_mod, only: theta0_at_xpoint`.

## Reproduction / evidence

**1. The push is load-bearing today, and `field_divB0.inp` is dead weight where it is read.**

`EXAMPLES/example_3` … `EXAMPLES/example_7` do not carry their own file at all — each `field_divB0.inp` there is a symlink to `INPUT/field_divB0.inp` (`git ls-files -s` shows mode `120000`, blob `5847dc7`). That single shared file has `iaxieq = 0` on line 13, yet `EXAMPLES/example_6` is the WEST case (`grid_kind = 4`, `g_file_filename = 'MHD_EQUILIBRIA/g_file_for_test_WEST'`). It only works because `SRC/tetra_grid_settings_mod.f90:90-97` sets `iaxieq_in = 1` for `grid_kind == 4` and `SRC/field_divB0.f90:66` overwrites the value just read. So the values in the file are already fiction for gfile, convexfile and iaxieq — which is exactly why lines 7 and 9 of `INPUT/field_divB0.inp` read `'see_tetra_grid_inp'`.

**2. The unconditional read costs the analytic path a dummy input file.**

`EXAMPLES/example_8` runs the analytic circular tokamak (`grid_kind = 5`, `EXAMPLES/example_8/tetra_grid.inp`), whose field is computed entirely in `SRC/field_analytic_circ_mod.f90` and needs no EFIT file, no coil file and no convex wall. It is the one example that ships a *real* `field_divB0.inp` (mode `100644`, byte-identical to `INPUT/field_divB0.inp`, 13 data lines including the dead site-specific path `'/itp/MooseFS/kasilov/MESH3D_31135odd/field.dat'`). It needs it because `SRC/field_divB0.f90:44` opens and lines 45-58 read that file unconditionally *before* the `if(ianalytic_circ.eq.1)` dispatch at `SRC/field_divB0.f90:69`:

```bash
cd /path/to/GORILLA
make                                     # -> BUILD/test_gorilla_main.x
mv EXAMPLES/example_8/field_divB0.inp /tmp/
cd EXAMPLES/example_8 && ./test_gorilla_main.x
```

Observed: `open(newunit=iunit, file='field_divB0.inp')` at line 44 uses the default `status='unknown'`, so an empty file is created and the first `read(iunit,*) ipert` hits end-of-file — the run aborts. Expected after the fix: example 8 runs to completion and produces `poincare_plot_phi_0_rphiz.dat` (the file `EXAMPLES/run_example.sh` checks for example 8) with no `field_divB0.inp` present at all.

Contrast with `EXAMPLES/example_1` and `EXAMPLES/example_2`, which ship **no** `field_divB0.inp` and pass CI. The reason is `SRC/tetra_physics_mod.f90:271`, which gates the first `call field(...)` on `grid_kind` ∈ {1, 2, 4, 5}; examples 1 and 2 are `grid_kind = 3` (VMEC) and never reach `field()`. (`EXAMPLES/example_7` is also `grid_kind = 3` and its symlinked `field_divB0.inp` is likewise never read — further evidence the file is not a universal requirement.)

## Behaviour (BDD)

```gherkin
Scenario: field_divB0 compiles without GORILLA's grid settings
  Given SRC/field_divB0.f90 after the refactor
  When I run "rg -n tetra_grid_settings_mod SRC/field_divB0.f90"
  Then there is no match
  And  "make" still links BUILD/test_gorilla_main.x

Scenario: EFIT run keeps taking filenames from tetra_grid.inp
  Given EXAMPLES/example_3 with tetra_grid.inp containing
        g_file_filename = 'MHD_EQUILIBRIA/g_file_for_test' and
        convex_wall_filename = 'MHD_EQUILIBRIA/convex_wall_for_test.dat'
  And   field_divB0.inp still carrying 'see_tetra_grid_inp' on lines 7 and 9
  When I run ./test_gorilla_main.x in EXAMPLES/example_3
  Then poincare_plot_phi_0_rphiz.dat is bit-identical to the pre-refactor run
  And  the string 'see_tetra_grid_inp' never reaches an open() call

Scenario: analytic circular tokamak needs no field_divB0.inp
  Given EXAMPLES/example_8 with grid_kind = 5 and field_divB0.inp deleted
  When I run ./test_gorilla_main.x
  Then the run completes with exit code 0
  And  poincare_plot_phi_0_rphiz.dat is written
  And  no file named field_divB0.inp is created in the working directory

Scenario: standalone use without any GORILLA namelist
  Given a driver that only calls read_field_input('field_divB0.inp') and field(...)
  And   field_divB0.inp line 7 gives 'MHD_EQUILIBRIA/g_file_for_test'
  And   field_divB0.inp line 9 gives 'MHD_EQUILIBRIA/convex_wall_for_test.dat'
  When the driver evaluates the field at (R,phi,Z) = (200.0, 0.0, 0.0) cm
  Then input_files::gfile and input_files::convexfile equal the values from the file
  And  no call to load_tetra_grid_inp() was required

Scenario: WEST equilibrium format still selected
  Given EXAMPLES/example_6 with grid_kind = 4
  And   its symlinked field_divB0.inp carrying iaxieq = 0 on line 13
  When load_tetra_grid_inp() runs
  Then field_mod::iaxieq equals 1 before the first call to field()
  And  the SOLEDGE3X-EIRENE run reproduces its current output within 1e-12 relative
```

## Implementation plan

1. **Create `SRC/field_input_mod.f90`.** New `module field_input_mod` holding the whole `field_divB0.inp` contract and nothing else. It must `use` only `input_files` (`SRC/utils_bdivfree.f90`), `field_mod`, `field_c_mod`, `field_eq_mod` and `inthecore_mod` — **never** `tetra_grid_settings_mod`, otherwise step 3 creates a circular file-level dependency in CMake. Contents:
   * three private module logicals `boole_gfile_set`, `boole_convexfile_set`, `boole_iaxieq_set`, all `= .false.`, recording which values a host code has pushed. (Three flags, not one, so a caller may push a subset.)
   * `subroutine set_field_input(gfile_in, convexfile_in, iaxieq_in)` with all three dummies `optional`: `character(*), intent(in), optional :: gfile_in, convexfile_in` and `integer, intent(in), optional :: iaxieq_in`. Each `present()` branch assigns the corresponding `input_files::gfile` / `input_files::convexfile` / `field_mod::iaxieq` and sets its flag.
   * `subroutine read_field_input(input_file)` with `character(*), intent(in), optional :: input_file`, modelled on libneo's `read_field_input`. It reads **all 13 positional lines** in the existing order (`ipert, iequil, ampl, ntor, cutoff, icftype, gfile, pfile, convexfile, fluxdatapath, nwindow_r, nwindow_z, iaxieq`), keeps the `err=` branch on the last line, and reads the three contested entries into local temporaries, copying each into its module target **only if the matching flag is `.false.`**. Open with `status='old', action='read'` (not the current default `status='unknown'`) so a missing file fails loudly instead of being silently created. Use `open(newunit=...)`, consistent with `210dffd`.
2. **Edit `SRC/field_divB0.f90`.** Delete the `use tetra_grid_settings_mod, ...` line (line 28) and the three assignments at lines 64-66 plus their comment at line 63. Replace the open/read block (lines 44-59) with `call read_field_input()`. Add `use field_input_mod, only: read_field_input`. Move the `if(ianalytic_circ.eq.1) then ... return` block (lines 69-76) **above** the `if(icall .eq. 0)` block so the analytic path never touches `field_divB0.inp`; keep the `icall = 1` assignment (line 43) inside the remaining guard. `character*1024 :: dummy` (line 38) becomes unused — remove it.
   * Safety note for the early return: on the analytic path `field_mod::iequil` is then never initialised, but it is only read at `SRC/field_divB0.f90:83`, i.e. after the return, so nothing regresses. `field_mod::ipert` is separately written by `SRC/tetra_physics_mod.f90:233` before the first `field()` call, so it stays defined.
3. **Edit `SRC/tetra_grid_settings_mod.f90`.** In `load_tetra_grid_inp` (line 81), after the `select case(grid_kind)` block at lines 90-97 that sets `iaxieq_in`/`ianalytic_circ`, add `use field_input_mod, only: set_field_input` and `call set_field_input(g_file_filename, convex_wall_filename, iaxieq_in)`. Also push `ianalytic_circ` into `field_mod` here rather than at `SRC/tetra_grid_mod.f90:139`, so it is set before *any* possible `field()` call. **Keep every variable `protected`** — nothing writes into this module, so the second comment's blocker disappears rather than being worked around.
4. **Edit `SRC/CMakeLists.txt`.** Add `field_input_mod.f90` to the `add_library(GORILLA STATIC ...)` list (lines 1-50), between `tetra_grid_settings_mod.f90` (line 6) and `field_divB0.f90` (line 25). Source order there is cosmetic — Ninja resolves Fortran module deps — but keep the list readable.
5. **Edit `SRC/tetra_grid_mod.f90`.** Drop the now-redundant `ianalytic_circ = 1` at line 139 once step 3 moved it; leave `call set_field_analytic_circ(...)` (lines 140-141) where it is, since it also seeds `field_eq_mod::rtf`/`btf`.
6. **Add `SRC/TESTS/test_field_input_mod.f90`** and register it in `SRC/TESTS/CMakeLists.txt` with `add_pfunit_ctest(test_field_input_mod TEST_SOURCES test_field_input_mod.f90 LINK_LIBRARIES GORILLA)`. Tests: (a) `read_field_input()` on the copied `INPUT/field_divB0.inp` sets `field_mod::ipert == 0`, `field_mod::iequil == 1`, `field_c_mod::ntor == 72`, `field_c_mod::icftype == 4` (lines 1, 2, 4, 6 of that file); (b) after `call set_field_input(gfile_in='MHD_EQUILIBRIA/g_file_for_test', iaxieq_in=0)` followed by `read_field_input()`, `trim(input_files::gfile)` still equals the pushed value and **not** `'see_tetra_grid_inp'`, while `trim(input_files::convexfile)` — not pushed — *does* equal `'see_tetra_grid_inp'`; (c) `read_field_input('field_divB0.inp')` with the explicit argument behaves identically to (a). Use `trim()` on both sides of every string assertion — `input_files::gfile` is `character*1024`. Extend the `file(COPY ...)` line at `SRC/TESTS/CMakeLists.txt:46` to also copy `../../INPUT/field_divB0.inp`.
7. **Extend `SRC/TESTS/test_tetra_grid_settings_mod.f90`** with an assertion that after `load_tetra_grid_inp()` (already invoked in the `@before` hook at line 18) `trim(input_files::gfile)` equals `'MHD_EQUILIBRIA/g_file_for_test'` — the value in `INPUT/tetra_grid.inp`, which the existing test already asserts for `g_file_filename` at line 38. This pins the push direction.
8. **Delete `EXAMPLES/example_8/field_divB0.inp`** once step 2 lands (it is a real file, not a symlink, so deleting it affects no other example). Update `INPUT/field_divB0.inp` lines 7 and 9: replace `'see_tetra_grid_inp'` with the real defaults `'MHD_EQUILIBRIA/g_file_for_test'` / `'MHD_EQUILIBRIA/convex_wall_for_test.dat'` and add a comment that GORILLA overrides these from `tetra_grid.inp`. Examples 3-7 symlink to this file, so the edit propagates to all of them at once.
9. **Downstream, separate PR: `itpplasma/GORILLA_APPLETS`.** It builds against this `SRC/` tree and ships eight `field_divB0.inp` files (`EXAMPLES/MONO_ENERGETIC_TRANSPORT/`, `EXAMPLES/POINCARE_INVARIANT/`, `EXAMPLES/anomalous_transport/`, `EXAMPLES/boltzmann_test/`, `EXAMPLES/divertor_heat_loads_lambda_0.7/`, `EXAMPLES/field_line_tracing/`, `EXAMPLES/helical_core/`, `EXAMPLES/self_consistent_electric_field/`). The positional format is unchanged by this plan, so no APPLETS change is *required*; verify by running one APPLETS example before merging here.
10. **Follow-up, not this PR.** Same treatment for `new_vmec_stuff_mod` (`SRC/canonical_coordinates_mod.f90:11`, alias `netcdffile => netcdf_filename`) and `SRC/field_line_integration_for_SYNCH.f90:30` (`theta0_at_xpoint`). Split those into an atomic issue "Remove `tetra_grid_settings_mod` imports from libneo-lineage modules" so this PR stays reviewable.

## Acceptance criteria

- [ ] `rg -n "tetra_grid_settings_mod" SRC/field_divB0.f90` returns nothing.
- [ ] `make` succeeds from a clean tree and produces `BUILD/test_gorilla_main.x`.
- [ ] `./build_coverage.sh` — the nine existing ctests in `SRC/TESTS/CMakeLists.txt` all pass, plus the new `test_field_input_mod`.
- [ ] `cd EXAMPLES/example_8 && ./test_gorilla_main.x` runs to completion with `field_divB0.inp` absent, and writes `poincare_plot_phi_0_rphiz.dat`.
- [ ] Examples 1-7 run under the CI recipe (`.github/workflows/Ubuntu.yml:72-74`: `for i in 1 2 3 4 5 6 7; do (cd EXAMPLES/example_$i && ./test_gorilla_main.x); done`), and each example's expected outputs — the per-example lists in `EXAMPLES/run_example.sh` — match the pre-refactor run to `1e-12` relative. The cases that actually exercise `gfile`/`convexfile`/`iaxieq` are the `grid_kind` ∈ {1, 2, 4, 5} ones: examples 3, 4, 5 (`grid_kind = 2`), 6 (`grid_kind = 4`) and 8 (`grid_kind = 5`). Concretely: `poincare_plot_phi_0_rphiz.dat` for examples 3, 4 and 8; `poincare_plot_phi_0_rphiz_trapped.dat` and `full_orbit_plot_rphiz_trapped.dat` for example 5; `full_orbit_plot_rphiz_passing.dat` for example 6.
- [ ] A short standalone driver that calls only `read_field_input()` and `field(...)` links against the GORILLA static library without any `tetra_grid.inp` present.
- [ ] `SRC/tetra_grid_settings_mod.f90` still declares every module variable `public, protected` (diff shows no removed `protected` keyword).

## References

No specific literature reference; this is an implementation detail internal to the code. The relevant prior art is in-org rather than published: libneo's `src/magfie/field_divB0.f90` already implements the target factoring (`read_field_input` with an optional path, zero host-code imports), and `libneo/src/magfie/input_files.f90:4` plus `libneo/src/vmec_support/canonical_coordinates_mod.f90:12` show the settable-module-variable convention this repo's copies diverged from. The GORILLA method itself is M. Eder et al., "Quasi-geometric integration of guiding-centre orbits in piecewise linear toroidal fields", *Physics of Plasmas* **27**, 122508 (2020), doi:10.1063/5.0022117 — cited in the header of `SRC/test_gorilla_main.f90:26-29`.

## Scope and risk

Single PR: one new source file, four edited source files, one CMake list, two test files. The main risk is ordering: `set_field_input` must run before the first `field()` call on every path — `SRC/test_gorilla_main.f90:62` calls `load_tetra_grid_inp()` first, and so does `SRC/TESTS/test_orbit_timestep_gorilla.f90:21`, but GORILLA_APPLETS drivers must be checked too, which is why step 3 puts the push inside `load_tetra_grid_inp` rather than in `make_tetra_grid`. Second risk is the `status='unknown'` → `status='old'` change, which converts today's silent empty-file creation into a hard error for any workflow that was accidentally relying on it — that is the intended behaviour but should be called out in the PR description.

Related: `itpplasma/GORILLA#45` ("Support custom input file paths via command-line flag") becomes near-trivial for this file once `read_field_input(input_file)` takes an optional path, so land this first and reference it there.
